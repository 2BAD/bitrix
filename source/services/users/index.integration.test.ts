import { describe, expect, it } from 'vitest'
import Bitrix from './../../bitrix'

const WEBHOOK_URL = process.env['WEBHOOK_URL']

if (!WEBHOOK_URL) {
  throw Error('Integration tests require environmental variable `WEBHOOK_URL` to be set')
}

const { users } = Bitrix(WEBHOOK_URL)

describe('Users', () => {
  describe('fields', () => {
    it('should get all fields', async () => {
      const { result } = await users.fields()
      expect(result).toMatchSnapshot()
    })
  })

  describe('get', () => {
    it('should get all users', async () => {
      const { result } = await users.get()
      expect(result).toMatchSnapshot()
    })
    it('should get entry with ID = 1', async () => {
      const { result } = await users.get('1')
      expect(result).toMatchSnapshot()
    })
    it('should return an error in case non existent ID is requested', async () => {
      const { result } = await users.get('1001')
      expect(result).toEqual([])
    })
  })
})
