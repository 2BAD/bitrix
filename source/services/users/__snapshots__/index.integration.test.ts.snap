// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users fields should get all fields 1`] = `
Object {
  "ACTIVE": "Активность",
  "EMAIL": "E-Mail",
  "ID": "ID",
  "LAST_NAME": "Фамилия",
  "NAME": "Имя",
  "PERSONAL_BIRTHDAY": "Дата рождения",
  "PERSONAL_CITY": "Город проживания",
  "PERSONAL_COUNTRY": "Страна",
  "PERSONAL_FAX": "Факс",
  "PERSONAL_GENDER": "Пол",
  "PERSONAL_ICQ": "ICQ",
  "PERSONAL_MOBILE": "Личный мобильный",
  "PERSONAL_PAGER": "Пейджер",
  "PERSONAL_PHONE": "Личный телефон",
  "PERSONAL_PHOTO": "Фотография",
  "PERSONAL_PROFESSION": "Профессия",
  "PERSONAL_STATE": "Область / край",
  "PERSONAL_STREET": "Улица проживания",
  "PERSONAL_WWW": "Домашняя страничка",
  "PERSONAL_ZIP": "Почтовый индекс",
  "SECOND_NAME": "Отчество",
  "UF_DEPARTMENT": "Подразделения",
  "UF_DISTRICT": "Район",
  "UF_FACEBOOK": "Facebook",
  "UF_INTERESTS": "Интересы",
  "UF_LINKEDIN": "LinkedIn",
  "UF_PHONE_INNER": "Внутренний телефон",
  "UF_SKILLS": "Навыки",
  "UF_SKYPE": "Skype",
  "UF_TWITTER": "Twitter",
  "UF_WEB_SITES": "Другие сайты",
  "UF_XING": "Xing",
  "USER_TYPE": "USER_TYPE",
  "WORK_COMPANY": "Компания",
  "WORK_PHONE": "Телефон компании",
  "WORK_POSITION": "Должность",
}
`;

exports[`Users get should get all users 1`] = `
Array [
  Object {
    "ACTIVE": true,
    "EMAIL": "2bad/bitrix@2bad.me",
    "ID": "1",
    "LAST_NAME": "McDickingson",
    "NAME": "Cunt",
    "PERSONAL_BIRTHDAY": "2019-09-05T03:00:00+03:00",
    "PERSONAL_CITY": "Havana",
    "PERSONAL_COUNTRY": null,
    "PERSONAL_FAX": null,
    "PERSONAL_GENDER": "M",
    "PERSONAL_ICQ": null,
    "PERSONAL_MOBILE": "+1 111 111-222",
    "PERSONAL_PAGER": null,
    "PERSONAL_PHONE": null,
    "PERSONAL_PHOTO": "https://cdn-ru.bitrix24.ru/b11676204/main/e00/e00e2d51b12dd3e5a433d3ba8a9bdeb0/1ed0f98db027bb4f2ee53a9433200a40.png",
    "PERSONAL_PROFESSION": null,
    "PERSONAL_STATE": null,
    "PERSONAL_STREET": null,
    "PERSONAL_WWW": "https://long-dong-between.us",
    "PERSONAL_ZIP": null,
    "SECOND_NAME": "",
    "UF_DEPARTMENT": Array [
      1,
    ],
    "UF_DISTRICT": null,
    "UF_FACEBOOK": null,
    "UF_INTERESTS": null,
    "UF_LINKEDIN": null,
    "UF_PHONE_INNER": "20",
    "UF_SKILLS": null,
    "UF_SKYPE": "sexually.active.toyota.prius",
    "UF_TWITTER": null,
    "UF_WEB_SITES": null,
    "UF_XING": null,
    "USER_TYPE": "employee",
    "WORK_COMPANY": null,
    "WORK_PHONE": "+1 111 111-111",
    "WORK_POSITION": "Big Boss",
  },
]
`;

exports[`Users get should get entry with ID = 1 1`] = `
Array [
  Object {
    "ACTIVE": true,
    "EMAIL": "2bad/bitrix@2bad.me",
    "ID": "1",
    "LAST_NAME": "McDickingson",
    "NAME": "Cunt",
    "PERSONAL_BIRTHDAY": "2019-09-05T03:00:00+03:00",
    "PERSONAL_CITY": "Havana",
    "PERSONAL_COUNTRY": null,
    "PERSONAL_FAX": null,
    "PERSONAL_GENDER": "M",
    "PERSONAL_ICQ": null,
    "PERSONAL_MOBILE": "+1 111 111-222",
    "PERSONAL_PAGER": null,
    "PERSONAL_PHONE": null,
    "PERSONAL_PHOTO": "https://cdn-ru.bitrix24.ru/b11676204/main/e00/e00e2d51b12dd3e5a433d3ba8a9bdeb0/1ed0f98db027bb4f2ee53a9433200a40.png",
    "PERSONAL_PROFESSION": null,
    "PERSONAL_STATE": null,
    "PERSONAL_STREET": null,
    "PERSONAL_WWW": "https://long-dong-between.us",
    "PERSONAL_ZIP": null,
    "SECOND_NAME": "",
    "UF_DEPARTMENT": Array [
      1,
    ],
    "UF_DISTRICT": null,
    "UF_FACEBOOK": null,
    "UF_INTERESTS": null,
    "UF_LINKEDIN": null,
    "UF_PHONE_INNER": "20",
    "UF_SKILLS": null,
    "UF_SKYPE": "sexually.active.toyota.prius",
    "UF_TWITTER": null,
    "UF_WEB_SITES": null,
    "UF_XING": null,
    "USER_TYPE": "employee",
    "WORK_COMPANY": null,
    "WORK_PHONE": "+1 111 111-111",
    "WORK_POSITION": "Big Boss",
  },
]
`;
